pkg/apis/workflow/v1alpha1/version_types.go

```diff
@@ -10,3 +15,16 @@ type Version struct {
 	Compiler     string `json:"compiler" protobuf:"bytes,7,opt,name=compiler"`
 	Platform     string `json:"platform" protobuf:"bytes,8,opt,name=platform"`
 }
+
+var verRe = regexp.MustCompile(`^v(\d+)\.(\d+)\.(\d+)`)
+
+// BrokenDown returns the major, minor and release components
```

Comment:
```suggestion
Components returns the major, minor, and patch components
```

the name is incorrect in the comment; also missing "patch". 
tbh though, I think having names in comments as this codebase does is an anti-pattern since it is redundant.

I might also suggest naming this `MajorMinorPatch` which would be more specific than `Components`

Reply:
This is a [godoc func description](https://tip.golang.org/doc/comment#func). Or it would be if I hadn't messed it up. I write other code that linter enforces these on exports and I think they are good.

I like `MajorMinorPatch`, it's better.

Reply:
Is it convention to start with the function name?
Odd, that feels like an anti-pattern to me since you usually read the function signature first, including in rendered godoc. But if that's how it is 🤷

Reply:
See the [link above](https://tip.golang.org/doc/comment#func), that's the rule.

Revive's [exported](https://github.com/mgechev/revive/blob/05c4801f9be1c3727e640c5c314cdf0694fce956/RULES_DESCRIPTIONS.md#exported) linter would complain if it didn't.

Reply:
I'm not misunderstanding you. 

On [that page](https://tip.golang.org/doc/comment) it specifies (can't link to the sentence)

> As with packages (above) and funcs (below), doc comments for types start with complete sentences naming the declared symbol.

Also of note is the first section on packages where it notes

> As can be seen in this example, Go doc comments use complete sentences.

Reply:
Oh, you had linked to the "Functions" section before but this is actually outside of the functions section so I hadn't read that part 😅

Thanks for clarifying and direct quoting it so I could find and understand it!